app/apps/manifest.rb

```diff
@@ -49,36 +50,128 @@ def self.load_from_string(yaml)
     InvalidManifest.new(e)
   end
 
-  def defaults
-    {"name" => "", "description" => "", "category" => "", "subcategory" => "" , "icon" => "", "role" => "", "url" => ""}
-  end
-
+  # @param [Hash] opts A hash of the options in the manifest
+  # @option opts [String] :name The name of the application
+  # @option opts [String] :description The description of the application
+  # @option opts [String] :category The category of the application
+  # @option opts [String] :subcategory The subcategory of the application
+  # @option opts [String] :icon The icon used on the dashboard, optionally a Font Awesome tag
+  # @option opts [String] :role Dashboard categorization
+  # @option opts [String] :url An optional redirect URL
   def initialize(opts)
     raise InvalidContentError.new unless(opts && opts.is_a?(Hash))
```

Comment:
If you just checked for `opts.respond_to?(:to_h)` then you could get away with initializing a manifest from another manifest (similar to `Pathname`).

```ruby
def initialize(opts)
  raise InvalidContentError.new unless opts.respond_to?(:to_h)
  @manifest_options = opts.to_h.with_indifferent_access
end
```

Then you can do things like....

```ruby
m = Manifest.new(name: "Name")
h = {name: "Name"}

my_manifest = Manifest.new(m)
my_manifest = Manifest.new(h)
```